Framework/Algorithms/src/CreateFloodWorkspace.cpp

```diff
@@ -0,0 +1,207 @@
+#include "MantidAlgorithms/CreateFloodWorkspace.h"
+#include "MantidAPI/FileProperty.h"
+#include "MantidAPI/IFunction.h"
+#include "MantidAPI/MatrixWorkspace.h"
+#include "MantidAPI/MultipleFileProperty.h"
+#include "MantidAPI/Run.h"
+#include "MantidKernel/ArrayProperty.h"
+#include "MantidKernel/ConfigService.h"
+#include "MantidKernel/FacilityInfo.h"
+#include "MantidKernel/ListValidator.h"
+
+using namespace Mantid::Kernel;
+using namespace Mantid::API;
+
+namespace {
+namespace Prop {
+std::string const FILENAME("Filename");
+std::string const OUTPUT_WORKSPACE("OutputWorkspace");
+std::string const START_X("StartSpectrumIndex");
+std::string const END_X("EndSpectrumIndex");
+std::string const EXCLUDE("Exclude");
+std::string const BACKGROUND("Background");
+} // namespace Prop
+
+double const VERY_BIG_VALUE = 10000.0;
```

Comment:
How big is very big? I assume this is to divide by, right? Maybe infinity would be better, or some way of masking and propagating the mask for the excluded detectors?

Reply:
This way was suggested by the scientists. But I can make the value bigger.